lib/ExtUtils/MakeMaker/Locale.pm

```diff
@@ -47,7 +47,10 @@ sub _init {
             unless (defined &GetInputCP) {
                 eval {
                     require Win32;
-                    eval { Win32::GetConsoleCP() };
+                    eval {
+                        local $SIG{__WARN__} = sub {} if ( $] lt '5.014' ); # suppress deprecation warning for inherited AUTOLOAD of Win32::GetConsoleCP()
```

Comment:
This should be `if "$]" < '5.014'`

Reply:
Changed to `$] < 5.014` (numeric, but more similar to the use throughout the rest of the module).

I'll force push the result once it passes CI.

Reply:
Should it? `$]` is a string value, not a numeric value…

Reply:
I'm happy replace the comparison with whatever idiom is the best/most correct.

Looking at the repo, `$] < ...` is what is used throughout (except within my proposed changes):

```
lib/ExtUtils/MM_Any.pm:1093:        my $p2m = sprintf <<'CMD', $section, $] > 5.008 ? " -u" : "";
lib/ExtUtils/MM_Cygwin.pm:89:        if( $] >= 5.006002 ) {
lib/ExtUtils/MM_NW5.pm:195:                              $] =~ /(\d)\.(\d{3})(\d{2})/;
lib/ExtUtils/MM_Unix.pm:2918:    if ( $pod_encoding and !( $] < 5.008 or !$Config{useperlio} ) ) {
lib/ExtUtils/MM_Unix.pm:3197:    if ($] >= 5.008) {
lib/ExtUtils/MM_VMS.pm:208:                push(@cand,$name) if $name =~ /^[\w\-\$]+$/;
lib/ExtUtils/MM_VMS.pm:220:        if ($name =~ /^[\w\-\$]+$/) {
lib/ExtUtils/MakeMaker.pm:401:    push @Get_from_Config, qw( vendorarchexp vendorlibexp ) if $] >= 5.006;
lib/ExtUtils/MakeMaker.pm:537:        !$self->{MIN_PERL_VERSION} or $self->{MIN_PERL_VERSION} <= $]
lib/ExtUtils/MakeMaker.pm:547:            die sprintf <<"END", $self->{MIN_PERL_VERSION}, $];
lib/ExtUtils/MakeMaker.pm:555:                $self->{MIN_PERL_VERSION}, $];
lib/ExtUtils/MakeMaker.pm:593:          $pr_version = $];
lib/ExtUtils/MakeMaker.pm:1269:        utf8::encode $to_write if !$CAN_DECODE && $] > 5.008;
lib/ExtUtils/MakeMaker/Locale.pm:51:                        local $SIG{__WARN__} = sub {} if ( $] lt '5.014' ); # suppress deprecation warning for inherited AUTOLOAD of Win32::GetConsoleCP()
lib/ExtUtils/MakeMaker/version.pm:38:        if ($] >= 5.009000) {
lib/ExtUtils/MakeMaker/version/vpp.pm:937:	if ( $] ge 5.008_001 ) {
lib/ExtUtils/MakeMaker/version/vpp.pm:941:	elsif ( $] ge 5.006_000 ) {
lib/ExtUtils/MakeMaker/version/vpp.pm:976:    if ( exists $INC{"$class.pm"} and not %{"$class\::"} and $] >= 5.008) {
lib/ExtUtils/MakeMaker/version/vpp.pm:985:	local $^W if $] <= 5.008;
lib/ExtUtils/MakeMaker/version/vpp.pm:992:	    my $msg =  $] < 5.006
lib/ExtUtils/Mksymlists.pm:151:    unless($] >= 5.016) {
```

My preference would be use the same comparision and then change them all in another commit.

Thoughts?

Reply:
@Leont it is a string value, but the correct way to compare it is numeric (or creating version.pm objects).

Reply:
What would go wrong the other way? This is rather counter-intuitive to me.

Reply:
With carefully constructed strings nothing should go wrong either way, but it is in principle a numeric comparison - it should still work if your version string is '005.008' for example

Reply:
`$]` should be converted to a string first, and then compared numerically. Zefram wrote about this a few years back documenting the various issues that can arise when compared otherwise.

I prepared a patch to fix this everywhere, in #333.

Reply:
@karenetheridge , What your timeline for merging #333? It causes a (minor) conflict with this PR, so I rebased on top of it. I'll push the updated branch (which uses `"$]" > ...`) when #333 is in master.